app/Http/Controllers/DinnerformController.php

```diff
@@ -0,0 +1,160 @@
+     */
+    public function show($id)
+    {
+        $dinnerform = Dinnerform::findOrFail($id);
+
+        if ($dinnerform->isCurrent()) {
+            return Redirect::away($dinnerform->url);
+        } else {
+            Session::flash("flash_message", "Sorry, you can't order anymore, food is already on its way");
+            return Redirect::route('homepage');
+        }
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function edit($id)
+    {
+        $dinnerformCurrent = Dinnerform::findOrFail($id);
+        $dinnerformList = Dinnerform::all()->sortByDesc('end');
+
+        return view('dinnerform.admin', ['dinnerformCurrent' => $dinnerformCurrent, 'dinnerformList' => $dinnerformList]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function update(Request $request, $id)
+    {
+
+        $dinnerform = Dinnerform::findOrFail($id);
+
+        $changed_important_details = $dinnerform->start->timestamp != strtotime($request->start) || $dinnerform->end->timestamp != strtotime($request->end) || $dinnerform->restaurant != $request->restaurant;
+
+        $dinnerform->restaurant = $request->restaurant;
+        $dinnerform->start = strtotime($request->start);
+        $dinnerform->end = strtotime($request->end);
+        $dinnerform->description = $request->description;
+
+        if ($dinnerform->end < $dinnerform->start) {
+            Session::flash("flash_message", "You cannot let the dinnerform close before it opens.");
+            return Redirect::back();
+        }
+
+        $dinnerform->save();
+
+        if ($changed_important_details) {
+            Session::flash("flash_message", "Your dinner form for '" . $dinnerform->restaurant . "' has been saved. You updated some important information. Don't forget to update your participants with this info!");
+        } else {
+            Session::flash("flash_message", "Your dinner form for '" . $dinnerform->restaurant . "' has been saved.");
+        }
+
+        return Redirect::route('dinnerform::add');
+
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $dinnerform = Dinnerform::findOrFail($id);
+
+        Session::flash("flash_message", "The dinner form for '" . $dinnerform->restaurant . "' has been deleted.");
+
+        if(URL::previous() != route('dinnerform::edit', ['id' => $dinnerform->id])) {
+            $dinnerform->delete();
+            return Redirect::back();
+        } else {
+            $dinnerform->delete();
+            return Redirect::route('dinnerform::add');
+        }
```

Comment:
The avoid code duplication, place the `$dinnerform->delete()` outside of this if else block